lib/utils.js

```diff
@@ -1,22 +1,55 @@
+/**
+ * Breaks down blobs (strings) in half until they fit in
+ * maximum length.
+ * Pass a maximum string length and an optional encoder (
+ * will ensure that even if the string gets encoded, the
+ * resulting string will be smaller than the maximum length),
+ * and the function will give you back another function, which
+ * you can use to break down a blob (string) into an array
+ * of strings.
+ * @param {number} maxLength 
+ * @param {(string) => string} (encode)
+ * @returns {(string) => Array<string>}
+ */
+const blobSplitter = (maxLength, encode = (e) => e) => (blob) => {
+  if (encode(blob).length <= maxLength || blob.length <= 1) {
+    return [blob];
+  } else {
+    const split = blobSplitter(maxLength, encode);
+    const limit = Math.floor(blob.length / 2);
+    const firstChunk = blob.substr(0, limit);
+    const secondChunk = blob.substr(limit);
+    return [...split(firstChunk), ...split(secondChunk)];
+  }
+};
+
 /**
  * Helper to split string by length
- * @param str
- * @param len
- * @returns {Array}
+ * @param str String ot be divided
+ * @param maxLength Maximum length of the string
+ * @param {(string) => string} (encode) An optional function that is
+ *        used to encode the chunks. If provided, the chunkString
+ *        function will ensure that the chunks are smaller than the
+ *        maxLength, even if the encoded result is larger than the
+ *        original string
+ * @returns {Array<string>}
  */
-const chunkString = function (str, len) {
-  const size   = Math.ceil(str.length / len);
-  const ret    = new Array(size);
-  let offset = null;
-
-  for (let i = 0; i < size; i++) {
-    offset = i * len;
-    ret[i] = str.substring(offset, offset + len);
+const chunkString = (str, maxLength, encode = (e) => e) => {
+  const chunkedSize = Math.ceil(str.length / maxLength);
+  const chunked = new Array(chunkedSize);
+  
+  for (let i = 0; i < chunkedSize; i++) {
+    chunked[i] = str.substr(i * maxLength, maxLength);
   }
 
-  return ret;
+  const splitBlobs = blobSplitter(maxLength, encode);
+
```

Comment:
Since this code will run in the browser, we should probably limit the amount of ES6 that is used. 
Things like arrow functions are still not supported by IE11:  https://caniuse.com/#feat=arrow-functions
map and reduce should be ok.  Same with desctructuring.

Reply:
A nice alternative to writing ES5 is using Babel/Webpack. Is adding that within the scope of this sprint?

Reply:
We are using Webpack already. I don't think ES6 code will be a problem